packages/server/lib/browsers/webkit-automation.ts

```diff
@@ -0,0 +1,163 @@
+import _ from 'lodash'
+
+import type playwright from 'playwright-webkit'
+
+export type CyCookie = Pick<chrome.cookies.Cookie, 'name' | 'value' | 'expirationDate' | 'hostOnly' | 'domain' | 'path' | 'secure' | 'httpOnly'> & {
+  // use `undefined` instead of `unspecified`
+  sameSite?: 'no_restriction' | 'lax' | 'strict'
+}
+
+type CookieFilter = {
+  name: string
+  domain: string
+}
+
+function convertSameSiteExtensionToCypress (str: CyCookie['sameSite']): 'None' | 'Lax' | 'Strict' | undefined {
+  return str ? ({
+    'no_restriction': 'None',
+    'lax': 'Lax',
+    'strict': 'Strict',
+  })[str] : str as any
+}
+
+const normalizeGetCookieProps = (cookie: any): CyCookie => {
+  if (cookie.expires === -1) {
+    delete cookie.expires
+  }
+
+  // Use expirationDate instead of expires 🤷‍♀️
+  cookie.expirationDate = cookie.expires
+  delete cookie.expires
+
+  if (cookie.sameSite === 'None') {
+    cookie.sameSite = 'no_restriction'
+  }
+
+  return cookie as CyCookie
+}
+
+const normalizeSetCookieProps = (cookie: CyCookie): playwright.Cookie => {
+  return {
+    name: cookie.name,
+    value: cookie.value,
+    path: cookie.path,
+    domain: cookie.domain,
+    secure: cookie.secure,
+    httpOnly: cookie.httpOnly,
+    expires: cookie.expirationDate!,
+    sameSite: convertSameSiteExtensionToCypress(cookie.sameSite)!,
```

Comment:
```suggestion
    sameSite: convertSameSiteExtensionToCypress(cookie.sameSite),
```

Seems `!` isn't necessary - at least, no error in my IDE.

Reply:
Playwright's `Cookie` type requires a value for SameSite, although this seems to work fine for the basic purpose of getting Cypress running, there is a test failure related to SameSite that's probably caused by this, which I'll fix in #23232 

I get this error in vscode:

```
Type '"None" | "Lax" | "Strict" | undefined' is not assignable to type '"None" | "Lax" | "Strict"'.
  Type 'undefined' is not assignable to type '"None" | "Lax" | "Strict"'.
```